king/king.wdl

```diff
@@ -0,0 +1,425 @@
+	Boolean defVariantInclude = defined(variant_include_file)
+	
+	command {
+		set -eux -o pipefail
+
+		echo "Generating config file"
+		# noquotes = argument has no quotes in the config file to match CWL config EXACTLY
+		python << CODE
+		import os
+
+		f = open("gds2bed.config", "a")
+
+		f.write('gds_file "~{gds_file}"\n')
+
+		if "~{defVariantInclude}" == "true":
+			f.write('variant_include_file "~{variant_include_file}"\n')
+
+		# check for empty string
+		if "~{bed_file}" != "":
+			f.write('bed_file "~{bed_file}"\n')
+		else:
+			f.write('bed_file "~{basename(gds_file, ".gds")}"\n')
+
+		if "~{defSampleInclude}" == "true":
+			f.write('sample_include_file "~{sample_include_file}"\n')
+
+		f.close()
+		CODE
+
+		echo "Calling R script gds2bed.R"
+		Rscript /usr/local/analysis_pipeline/R/gds2bed.R gds2bed.config
+	}
+
+	runtime {
+		cpu: cpu
+		docker: "uwgac/topmed-master@sha256:0bb7f98d6b9182d4e4a6b82c98c04a244d766707875ddfd8a48005a9f5c5481e"
+		disks: "local-disk " + final_disk_dize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		File processed_bed = glob("*.bed")[0]
+		File processed_bim = glob("*.bim")[0]
+		File processed_fam = glob("*.fam")[0]
+		File config_file = "gds2bed.config"
+	}
+
+}
+
+# [2] plink_make_bed -- process bed file through plink
+
+# Add java script from cwl?
+
+task plink_make_bed {
+	input {
+		File bedfile
+		File bimfile
+		File famfile
+		
+		# runtime attributes
+		Int addldisk = 5
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+
+	# Estimate disk size required
+	Int bed_size = ceil(size(bedfile, "GB"))
+	Int final_disk_dize = bed_size + addldisk
+
+	
+	command {
+		set -eux -o pipefail
+
+		echo "Generating bash executable file"
+		python << CODE
+		import os
+
+		f = open("plink_cmd.sh", "a")
+
+		f.write('#!/bin/bash\n')
+		f.write('plink --make-bed --bfile ' + '.'.join("~{bimfile}".split('.')[:-1]) \
+		+ ' --bed ' + "~{bedfile}" + ' --fam ' + "~{famfile}" + ' --out ' \
+		+ os.path.basename('.'.join(("~{bedfile}").split('.')[:-1])) + '_recode\n')
+	
+		CODE
+
+		echo "Calling command through executable bash file for plink"
+		chmod u+x plink_cmd.sh
+		./plink_cmd.sh
+
+	}
+
+	runtime {
+		cpu: cpu
+		docker: "uwgac/topmed-master@sha256:0bb7f98d6b9182d4e4a6b82c98c04a244d766707875ddfd8a48005a9f5c5481e"
+		disks: "local-disk " + final_disk_dize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		File bed_file = glob("*_recode.bed")[0]
+		File bim_file = glob("*.bim")[0]
+		File fam_file = glob("*.fam")[0]
+	}
+
+}
+
+# [3] king_ibdseg -- Use KING --ibdseg method to estimate kinship coefficients,
+# and returns results for pairs of related samples.
+
+task king_ibdseg {
+	input {
+		File bed_file
+		File bim_file
+		File fam_file
+		String? out_prefix
+		
+		# runtime attributes
+		Int addldisk = 5
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+
+	# Estimate disk size required
+	Int bed_size = ceil(size(bed_file, "GB"))
```

Comment:
Based on the truth file you uploaded, I think there's a decent chance the seg file will be pretty big when run on "real" data. As with gds2bed, try to gauge the size of the output file based on the size of the input bed and account for that in your final_disk_size calc.